storage/raft_transport.go

```diff
@@ -192,8 +192,18 @@ func (t *RaftTransport) processQueue(nodeID roachpb.NodeID) {
 		return
 	}
 	client := NewMultiRaftClient(conn)
+
+	var wg sync.WaitGroup
 	ctx, cancel := context.WithCancel(context.TODO())
-	defer cancel()
+	defer func() {
+		// After pending Raft requests have finished, cancel the Context to free
+		// gRPC resources.
+		wg.Wait()
+		if log.V(1) {
+			log.Infof("cancelling context")
+		}
+		cancel()
+	}()
```

Comment:
This seems odd. From looking at the code I would think we are trying to explicitly cancel inflight raft requests when we return. What code path is being taken that we're returning from this method but don't want to cancel inflight raft requests?